Add KATANA_LOG_DEBUG_ABORT and replace abort and KATANA_ABORT

arthurp opened the pull request:
This is a step towards solving: KatanaGraph/katana-enterprise#698

gurbinder533 commented on lonestardist/bc/bc_level.cpp:
```diff
@@ -235,8 +245,8 @@ struct ForwardPass {
     do {
       _dga.reset();
 
-#ifdef __GALOIS_HET_CUDA__
```
Any particular reason for moving this conditional out?

arthurp committed:
Add KATANA_LOG_DEBUG_ASSERT, etc. Abstract application abort.

arthurp committed:
Change assert and KATANA_ASSERT to KATANA_LOG_ macros.

assert(...) -> KATANA_LOG_DEBUG_ASSERT(...)
assert(... && "...") -> KATANA_LOG_DEBUG_VASSERT(..., "...")
KATANA_ASSERT(...) ->  KATANA_LOG_ASSERT(...)
KATANA_ASSERT(..., "...") ->  KATANA_LOG_VASSERT(..., "...")

Uses of assert in tests are not changed.

tylershunt reviewed (approved):
Nice!

ddn0 reviewed (approved):
LGTM!
